Userland/Libraries/LibVideo/PlaybackManager.cpp

```diff
@@ -68,22 +68,10 @@ void PlaybackManager::set_playback_status(PlaybackStatus status)
             m_present_timer->stop();
         }
 
-        m_main_loop.post_event(*this, make<PlaybackStatusChangeEvent>(status, old_status));
+        m_main_loop.post_event(m_event_handler, make<PlaybackStatusChangeEvent>(status, old_status));
     }
 }
 
-void PlaybackManager::event(Core::Event& event)
-{
-    if (event.type() == DecoderErrorOccurred) {
-        auto& error_event = static_cast<DecoderErrorEvent&>(event);
-        VERIFY(error_event.error().category() != DecoderErrorCategory::EndOfStream);
-        set_playback_status(PlaybackStatus::Corrupted);
```

Comment:
I'm not seeing it in this PR, but do we still set the playback status of the PlaybackManager to corrupted if a decoder error occurs? that would have to happen from the widget that's its parent right?

Reply:
That was actually already handled by on_decoder_error:

https://github.com/SerenityOS/serenity/blob/851807a8ab85bc2d3c24346be3c4d0e818bd979f/Userland/Libraries/LibVideo/PlaybackManager.cpp#L117

That was part of some previous changes, which at the time I hadn't realized enabled me to ditch the Object hierarchy.